cEP-0022.md

```diff
@@ -0,0 +1,461 @@
+`inspect` module at the meta class `bearclass`. The operations performed
+at the `bearclass` will include populating classes (referred to as `Quickclass`
+class from here onwards) which will store metadata
+about the type of settings. Manual checks will be needed for determining which
+settings are Type 3 settings.
+
+## Post-Processing of settings
+
+Many bear settings of different bears achieve the same function. We will
+either pick randomly one of the settings or bias towards a particular
+setting if we see certain issues arising for eg.. `PEP8Bear` failing
+to do line length checks in some cases is a known issue.
+The settings not required will
+be removed from `Quickclass`. Manual sorting needs to be performed to
+identify such settings.
+
+So we are applying restrictions on bear settings using Type Annotations
+while distinction of bear settings is done at the meta-class.
+
+# General Operations
+
+These consist of developing classes and method which will be used both by the
+**Brute Force** and when we improve upon it.
+
+- coala will be ran over and over again on the given file dict, using
+  `Quickclass` for correct detection of type of
+  setting and then guessing its value the appropriate way as described above.
+- Undetermined values for Type 1, Type 2 and Type 3 settings go into the
+  `.quickfile`
+  while successful Type 1 combinations of setting values go into
+  `QUICKINFO.yaml`. The data generated by *QuickstartBear* goes into
+  `QUICKDATA.yaml`
+- *"Green"* values for settings are added to `.coafile`
+
+# Acting upon the newly available metadata
+The metadata has been generated in the following files:
+- `.quickfile`
+- `QUICKDATA.yaml`
+- `QUICKINFO.yaml`
+and the following classes:
+- `Quickclass`
+
+## `.quickfile`
+Will have 3 sections as created in `.coafile`
+- **[severe]**
+  This will contain the **Type 1** and **Type 3** dropped settings along with their
+  values and weights for each value.
+- **[amenable]**
+  This will contain the **Type 2** dropped settings along with their weights and
+  values.
+- **[permanent]**
+  This will contain values to settings that are obtained when the user answers
+  the questions prepared by `coala-quickstart`. Special string may be used
+  to represent confirmed settings that need to be dropped.
+
+`coala-quickstart` will always ask at the end of the run
+whether the user is interested in answering
+some questions which will lead to more secure creation of config files. If the
+user choses to answer them, `coala-quickstart` will provide the user with a
+bunch of questions from the sections **[severe]** an **[amenable]** asking
+them whether an inconsistency detected is a mistake in the code-base or there
+are no style rules followed by the project regarding the particular setting.
+The answered bear-settings
+will be moved to the **[permanent]** section.
+
+These set of questions can be invoked directly without a `--green-mode` run
```

Comment:
yep, always ask to ask those questions in the end

Reply:
is that behaviour unwanted?

Reply:
We'll have to see if it produces better results than the current one, the default mode ideally should be just enough accurate and just enough fast so that it does not annoy the users.